awswrangler/s3/_write_text.py

```diff
@@ -474,6 +480,16 @@ def to_csv(  # pylint: disable=too-many-arguments,too-many-locals,too-many-state
             pd_kwargs.pop("compression", None)
 
         df = df[columns] if columns else df
+
+        columns_types: Dict[str, str] = {}
+        partitions_types: Dict[str, str] = {}
+        if (database is not None) and (table is not None):
+            columns_types, partitions_types = _data_types.athena_types_from_pandas_partitioned(
+                df=df, index=index, partition_cols=partition_cols, dtype=dtype, index_left=True
+            )
+            if schema_evolution is False:
+                _check_schema_changes(columns_types=columns_types, table_input=catalog_table_input, mode=mode)
+
```

Comment:
Why was it necessary to move this block outside the existing `if` condition for database and table on line 499?

Reply:
Hahah, yes, I wanted to do it before `_to_dataset` call which in some cases will send a request to delete the files; and right after `df = df[columns] if columns else df` which is right when we form a "final" dataframe that we should check for schema evolution.